Move tax rate setup to the Settings page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Project thread: pcTzPl-2Cf-p2

This PR implements the **📌 Move tax setting to main setting page** task of the project.

- The tax rate setup has been hidden by #2542 in the onboarding flow, and it should remain hidden after moving the setup. Therefore, the tax rate setup is completely removed from the onboarding flow.
- The tax rate setup is moved from the Edit free listings page to the Settings page.

💡 Update for `src/Tracking/README.md` will be done by the conjunction PR for the feature branch.

### Screenshots:

![image](https://github.com/user-attachments/assets/adc05e99-98c7-46e9-90dc-14d1519dc16c)

### Detailed test instructions:

1. Set up the country of the store address to the US via WooCommerce Settings or set up the audience countries of this plugin to include the US.
2. Go to the Edit free listings page
   - Confirm the tax rate setup is not shown
   - Check if the editing function works
3. Go to the Settings page.
   - Check if the tax rate setup is shown
   - Open the Network tab of DevTool
   - Check if the tax rate setup can be saved
   - Inspect the Network tab to see if the requests are issued in the following order and if they are not in parallel:
      1. `wc/gla/mc/settings`
      2. `wc/gla/mc/settings/sync`
   - Change the country of the store address and audience countries to exclude the US
   - Reload the Settings webpage and check if the tax rate setup is not shown
4. View the E2E testing result: https://github.com/woocommerce/google-listings-and-ads/actions/runs/12593213104

### Changelog entry

> Update - Move tax rate setup from the Edit free listings page to the Settings page.

eason9487 committed:
Remove the `hideTaxRates` prop from the `SetupFreeListings` component and the related components and functions.

eason9487 committed:
Remove the tax rate setting from the `SetupFreeListings` component and the related components and functions.

eason9487 committed:
Remove the validation error message from `TaxRate` component and disable options during submission.

eason9487 committed:
Adjust `TaxRate` component to render children within its section wrapper and below its card.

eason9487 committed:
Add a new component to save and sync the tax rate setup individually.

eason9487 committed:
Add the tax rate setup to the Settings page.

eason9487 committed:
Remove unused `ConditionalSection` component.

eason9487 committed:
Update E2E tests.

puntope commented on tests/e2e/specs/settings/settings.test.js:
```diff
@@ -0,0 +1,99 @@
+			).not.toBeVisible();
+		} );
+
+		test( 'Update the setting', async () => {
```
```suggestion
		test( 'Should show the setup when selling to the US and can update the setting', async () => {
```

puntope commented:
WDYT about adding a confirmation message as feedback when the Setting save is completed.


https://github.com/user-attachments/assets/54709f1a-e780-4b73-b3ff-b740e3bbda28

puntope reviewed (approved):
Thanks @eason9487 for working on this. I didn't found any blocker so far. Just a recommendation about adding a message after saving taxes setting. Not sure if this was already discussed with Design. In any case not a strong blocker for me and can be done also in further PRs

- Tested the Taxes Section is not shown in Edit Free listings ✅ 
- Tested the Edit Free listings still work and can be saved ✅ 
- Tested the Taxes Section is not shown in Settings when neither US is the store base address nor a audience target country ✅
-  Tested the Taxes Section is shown in Settings when either US is the store base address or a audience target country ✅
- Tested you can save the taxes and it calls the API endpoints ✅ 
- UI general checks 
- Code & Tests checks 
- Additional Onboarding check

eason9487 committed:
Update the description for a test case in `tests/e2e/specs/settings/settings.test.js`

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2749#discussion_r1901736413

Co-authored-by: Miguel Pérez Pellicer <5908855+puntope@users.noreply.github.com>

eason9487 committed:
Add the confirmation notification after the tax rate setting is successfully saved and synced.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2749#issuecomment-2569183046

eason9487 commented:
Hi @puntope, thanks for the review!

> WDYT about adding a confirmation message as feedback when the Setting save is completed.

Good idea! Added in https://github.com/woocommerce/google-listings-and-ads/pull/2749/commits/fdae02a8d50106f2589b741f985a9450e113be88.

https://github.com/user-attachments/assets/e9fedf5a-e9f4-4165-9d27-6dee8bcec7e6
